assembl/static2/js/app/components/administration/voteSession/gaugeForm.jsx

```diff
@@ -0,0 +1,130 @@
+// @flow
+import React from 'react';
+import { connect } from 'react-redux';
+import { I18n, Translate } from 'react-redux-i18n';
+import range from 'lodash/range';
+import { SplitButton, MenuItem, Radio } from 'react-bootstrap';
+import Helper from '../../common/helper';
+import { getEntryValueForLocale } from '../../../utils/i18n';
+import FormControlWithLabel from '../../common/formControlWithLabel';
+import NumberGaugeForm from './numberGaugeForm';
+import TextGaugeForm from './textGaugeForm';
+import {
+  updateGaugeVoteInstructions,
+  updateGaugeVoteNbTicks,
+  updateGaugeVoteIsNumber,
+  createGaugeVoteChoice,
+  deleteGaugeVoteChoice
+} from '../../../actions/adminActions/voteSession';
+
+type GaugeFormProps = {
+  id: string,
+  editLocale: string,
+  instructions: string,
+  nbTicks: number,
+  isNumberGauge: boolean,
+  choices: Object,
+  handleInstructionsChange: Function,
+  handleNbTicksSelectChange: Function,
+  handleNumberGaugeCheck: Function,
+  handleNumberGaugeUncheck: Function
+};
+
+const DumbGaugeForm = ({
+  id,
+  editLocale,
+  instructions,
+  nbTicks,
+  isNumberGauge,
+  choices,
+  handleInstructionsChange,
+  handleNbTicksSelectChange,
+  handleNumberGaugeCheck,
+  handleNumberGaugeUncheck
+}: GaugeFormProps) => (
+  <div className="gauges-vote-form">
+    <div className="flex margin-m">
+      <FormControlWithLabel
+        value={instructions}
+        label={I18n.t('administration.gaugeVoteInstructions')}
+        required
+        type="text"
+        onChange={handleInstructionsChange}
+      />
+      <Helper
+        helperUrl="/static2/img/helpers/helper6.png"
+        helperText={I18n.t('administration.helpers.gaugeVoteInstructions')}
+        additionalTextClasses="helper-text-only"
+      />
+    </div>
+    <div className="flex">
+      <label htmlFor={`dropdown-${id}`}>
+        <Translate value="administration.nbTicks" />
+      </label>
+      <Helper helperUrl="/static2/img/helpers/helper2.png" helperText={'administration.nbTicksHelper'} />
+    </div>
+    <SplitButton
+      title={nbTicks}
+      id={`dropdown-${id}`}
+      required
+      onSelect={(eventKey) => {
+        handleNbTicksSelectChange(eventKey, isNumberGauge, nbTicks);
+      }}
+    >
+      {range(10).map(value => (
+        <MenuItem key={`gauge-notch-${value + 1}`} eventKey={value + 1}>
+          {value + 1}
+        </MenuItem>
+      ))}
+    </SplitButton>
+    <div className="margin-m">
+      <Radio onChange={handleNumberGaugeUncheck} checked={!isNumberGauge}>
+        <Translate value="administration.textValue" />
+      </Radio>
+      <Radio onChange={handleNumberGaugeCheck} checked={isNumberGauge}>
+        <Translate value="administration.numberValue" />
+      </Radio>
+    </div>
+    {isNumberGauge && <NumberGaugeForm id={id} />}
+    {!isNumberGauge &&
+      choices.map((cid, index) => <TextGaugeForm key={`gauge-choice-${index}`} id={cid} index={index} editLocale={editLocale} />)}
```

Comment:
Why `TextGaugeForm`is not at the same "level" as `NumberGaugeForm`. I would move the `choices.map` in `TextGaugeForm` (when I read the component, I didn't understand why there is only one field)